### PR TITLE
resticprofile: 0.27.1 -> 0.28.0

### LINKED ISSUE #335966: Update request: resticprofile 0.27.0 → 0.28.0
- Package name: `resticprofile`
- Latest released version: 0.28.0

- Current version on the unstable channel: 0.27.0
- Current version on the stable/release channel: N/A

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

There is an old one for updating to `0.27.1` https://github.com/NixOS/nixpkgs/pull/330444

**Notify maintainers**

@TomaSajt 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/335966

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by TomaSajt** [comment]
@jloh I'd appreciate if you could test if it works for you

(I still haven't got around to actually learning to use the program, I just packaged it...)

**Comment by jloh** [comment]
@TomaSajt I've tested out the PR and builds successfully on my Mac running 14.6.1. Have tested the resulting binary and it works as expected with the changes of the new version I was expecting 👍 I don't know how to do a formal review but it all works as far as I can tell.

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 336074` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>resticprofile</li>
  </ul>
</details>

**Comment by JohnRTitor** [comment]
Build fails in checkphase weird

**Comment by JohnRTitor** [comment]
@ofborg build resticprofile

**Comment by jloh** [comment]
Is it possible these tests are failing because of the nix build env like they were in the last PR? https://github.com/NixOS/nixpkgs/pull/330502

**Comment by jloh** [comment]
Result of `nixpkgs-review pr 336074` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>resticprofile</li>
  </ul>
</details>

All works fine from my machine now and Borg is passing.

**Comment by DavHau** [comment]
Result of `nixpkgs-review pr 336074` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>resticprofile</li>
  </ul>
</details>

**Comment by jloh** [comment]
Thanks so much @Lassulus!

File: pkgs/by-name/re/resticprofile/package.nix [hunk 1]
```diff
@@ -9,15 +10,19 @@
   resticprofile,
 }:
 
-buildGoModule rec {
+let
+  # can be removed when the default go version is at least 1.23
+  buildGoModule' = buildGoModule.override { go = go_1_23; };
```
Comment (SuperSandro2000) [inline pkgs/by-name/re/resticprofile/package.nix:15]: Why wasn't buildGo123Module used instead?

Comment (TomaSajt) [reply pkgs/by-name/re/resticprofile/package.nix:15]: Well, I did not know it existed.

**Pull request merged**